### PR TITLE
[Shop][Product] Adding products to cart

### PR DESCRIPTION
**Pull request by Wojdylak**
| Q               | A
|-----------------|-----
| Branch?         | bootstrap-shop
| Bug fix?        | no
| New feature?    | no
| BC breaks?      | no
| Deprecations?   | no
| Related tickets | n/a
| License         | MIT

### CODE REVIEW

File: src/Sylius/Bundle/ShopBundle/Twig/Component/Product/SummaryComponent.php [hunk 1]
```diff
@@ -17,12 +17,14 @@
...
 use Sylius\Component\Product\Resolver\ProductVariantResolverInterface;
 use Sylius\TwigHooks\LiveComponent\HookableLiveComponentTrait;
+use Symfony\UX\LiveComponent\Attribute\AsLiveComponent;
 use Symfony\UX\LiveComponent\Attribute\LiveArg;
 use Symfony\UX\LiveComponent\Attribute\LiveListener;
 use Symfony\UX\LiveComponent\Attribute\LiveProp;
 use Symfony\UX\LiveComponent\DefaultActionTrait;
 use Symfony\UX\TwigComponent\Attribute\PostMount;
 
+#[AsLiveComponent]
```
Comment (mpysiak) [inline src/Sylius/Bundle/ShopBundle/Twig/Component/Product/SummaryComponent.php:27]: 🙃

**Review by mpysiak — APPROVED** [review]

File: src/Sylius/Bundle/ShopBundle/Twig/Component/Product/SummaryComponent.php (hunk 1 above)
Comment (Wojdylak) [reply src/Sylius/Bundle/ShopBundle/Twig/Component/Product/SummaryComponent.php:27]: :smiley:

File: src/Sylius/Bundle/ShopBundle/EventListener/CartItemRemoveListener.php [hunk 2]
```diff
@@ -11,20 +11,20 @@
...
 
-final readonly class OrderItemModifierListener
+final readonly class CartItemRemoveListener
 {
     public function __construct(private OrderModifierInterface $orderModifier)
     {
     }
 
-    public function removeOrderItem(GenericEvent $event): void
+    public function removeFromOrder(GenericEvent $event): void
```
Comment (TheMilek) [inline src/Sylius/Bundle/ShopBundle/EventListener/CartItemRemoveListener.php:27]:
```suggestion
    public function removeItemFromOrder(GenericEvent $event): void
```
🤔

File: src/Sylius/Bundle/ShopBundle/templates/cart/index/form/sections/general/items/body/item.html.twig [hunk 3]
```diff
@@ -1,5 +1,5 @@
 {% set item = hookable_metadata.context.item %}
 
 <td {{ sylius_test_html_attribute('cart-item-product' ) }}>
-    {% include '@SyliusShop/shared/product/info.html.twig' with {'variant': item.variant} %}
+    {{ component('sylius_shop:order_item:product_info', { item: item }) }}
```
Comment (TheMilek) [inline src/Sylius/Bundle/ShopBundle/templates/cart/index/form/sections/general/items/body/item.html.twig:4]: can we declare this component in hooks?

**Review by TheMilek — APPROVED** [review]

File: src/Sylius/Bundle/ShopBundle/Twig/Component/Product/AddToCartFormComponent.php [hunk 4]
```diff
@@ -13,75 +13,115 @@
...
-            $this->orderItem = $orderItem;
+        $this->eventDispatcher->dispatch(new GenericEvent($addToCartCommand), SyliusCartEvents::CART_ITEM_ADD);
+        $this->manager->persist($addToCartCommand->getCart());
+        $this->manager->flush();
 
-            $this->quantityModifier->modify($this->orderItem, 1);
-        }
+        FlashBagProvider
+            ::getFlashBag($this->requestStack)
+            ->add('success', 'sylius.cart.add_item');
```
Comment (GSadee) [inline src/Sylius/Bundle/ShopBundle/Twig/Component/Product/AddToCartFormComponent.php:111]:
```suggestion
        FlashBagProvider::getFlashBag($this->requestStack)
            ->add('success', 'sylius.cart.add_item')
        ;
```
or even
```suggestion
        FlashBagProvider::getFlashBag($this->requestStack)->add('success', 'sylius.cart.add_item');
```

File: src/Sylius/Component/Core/Factory/CartItemFactory.php [hunk 5]
```diff
@@ -44,6 +48,7 @@ public function createForProduct(ProductInterface $product): OrderItemInterface
         $variant = $this->variantResolver->getVariant($product);
         Assert::nullOrIsInstanceOf($variant, ProductVariantInterface::class);
         $cartItem->setVariant($variant);
+        $this->orderItemQuantityModifier->modify($cartItem, 1);
```
Comment (GSadee) [inline src/Sylius/Component/Core/Factory/CartItemFactory.php:51]: Why is this necessary? 🤔

**Review by GSadee — APPROVED** [review]
